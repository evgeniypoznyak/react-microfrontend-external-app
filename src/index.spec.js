import React from 'react';
import ReactDOM from 'react-dom';
import sinon from 'sinon';
import { render } from '@testing-library/react';

let div;
const APP_NAME = process.env.REACT_APP_NAME;
const methodMount = `render${APP_NAME}`;
const methodUnmount = `unmount${APP_NAME}`;
let AppWrapper;
let container;

beforeEach(() => {
  div = document.createElement('div');
  div.id = 'root';
  document.body.appendChild(div);
  AppWrapper = require('./index').AppWrapper;
  container = render(<AppWrapper />).container;
});

afterEach(() => {
  AppWrapper = null;
  container = null;
});

describe('index.js', () => {
  it('renders correctly', () => {
    expect(container).toMatchSnapshot();
  });

  it('Should attach react app into div with id=root ', () => {
    global[methodMount]('root');
    expect(container.childElementCount).toBeGreaterThan(0);
  });

  it('Should call unmountComponentAtNode if methodUnmount is called', async () => {
    const reactDOMStub = sinon.stub(ReactDOM, 'unmountComponentAtNode');
    global[methodUnmount]({ containerId: 'root' });
    expect(reactDOMStub.calledOnce).toBe(true);
  });
});
