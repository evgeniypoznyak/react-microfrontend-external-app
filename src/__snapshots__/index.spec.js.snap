// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.js renders correctly 1`] = `
<div>
  <div>
    Hi! I'm a external component from another server!
  </div>
</div>
`;
